src/s3ql/backends/swiftks.py

```diff
@@ -76,15 +79,22 @@ def _get_conn(self):
             tenant = None
             user = self.login
 
+        # We can optionally configure with tenant-name. If specified, it overrides tenant id from the login
+        tenant_name = self.options.get('tenant-name', None)
+
+        # We can configure with domain id and/or domain name. If both are specified, domain-name takes preference
         domain = self.options.get('domain', None)
-        if domain:
+        domain_name = self.options.get('domain-name', None)
+        if domain or domain_name:
             if not tenant:
                 raise ValueError("Tenant is required when Keystone v3 is used")
 
             # In simple cases where there's only one domain, the project domain
             # will be the same as the authentication domain, but this option
             # allows for them to be different
+            # We can configure with project-domain and/or project-domain-name. If both are specified, project-domain-name takes preference
             project_domain = self.options.get('project-domain', domain)
+            project_domain_name = self.options.get('project-domain-name', domain_name)
 
```

Comment:
I would throw an error in the case that the user provides both `domain` and `domain-name` or `project-domain` and `project-domain-name` and not silently ignore one of them.